src/plugins/themes.js

```diff
@@ -0,0 +1,84 @@
+"use strict";
+
+const fs = require("fs");
+const Helper = require("../helper");
+const colors = require("colors/safe");
+const path = require("path");
+const themes = new Map();
+
+module.exports = {
+	get: get,
+	fileName: fileName
+};
+
+fs.readdir("client/themes/", (err, builtInThemes) => {
+	if (err) {
+		return;
+	}
+	builtInThemes
+		.filter((theme) => theme.endsWith(".css"))
+		.map(makeLocalThemeObject)
+		.forEach((theme) => themes.set(theme.name, theme));
+});
+
+fs.readdir(Helper.getPackagesPath(), (err, packages) => {
+	if (err) {
+		return;
+	}
+	packages
+		.map(makePackageThemeObject)
+		.forEach((theme) => {
+			if (theme) {
+				themes.set(theme.name, theme);
+			}
+		});
+});
+
+function get() {
+	return Array.from(themes.values());
+}
+
+function fileName(module) {
+	if (themes.has(module)) {
+		return themes.get(module).filename;
+	}
+}
+
+function makeLocalThemeObject(css) {
+	const themeName = css.slice(0, -4);
+	return {
+		displayName: themeName.charAt(0).toUpperCase() + themeName.slice(1),
+		filename: path.join(__dirname, "..", "client", "themes", `${themeName}.css}`),
```

Comment:
Is that trailing `}` a typo?

Reply:
Hmmmm, yeah, it is. I didn't test local themes at the end, will sort this out.

Reply:
Weirdly, these still work fine. I assume `path.join` or `express.sendFile` filter it out. But I'll remove it, obviously.